src/farcasterd/trade_state_machine.rs

```diff
@@ -0,0 +1,1295 @@
+    /// sending SetOfferStatus to databased.
+    #[display("Make Offer")]
+    MakeOffer(MakeOffer),
+
+    /// TakerCommit state - transitions to SwapdLaunched once LaunchSwap is
+    /// received from walletd. Transition to SwapdLaunched triggers launch
+    /// swapd.
+    #[display("Taker Commit")]
+    TakerCommit(TakerCommit),
+
+    /// TakeOffer state - transitions to SwapdLaunched once LaunchSwap is
+    /// received from walletd. Transition to SwapdLaunched triggers launch
+    /// swapd.
+    #[display("Take Offer")]
+    TakeOffer(TakeOffer),
+
+    /// RestoringSwapd state - transitions to SwapdRunning once the various
+    /// Hello's are received. Transition to RestoringSwapd triggers
+    /// RestoreCheckpoint request to database.
+    #[display("Restoring Swapd")]
+    RestoringSwapd(RestoringSwapd),
+
+    /// SwapdLaunched state - transitions to SwapRunning once the various
+    /// Hello's are reiceved. Transition to SwapRunning triggers sending
+    /// TakeSwap to swapd
+    #[display("Swapd Launched")]
+    SwapdLaunched(SwapdLaunched),
+
+    /// SwapdRunning state - transitions to None once the SwapOutcome is received
+    /// from its swapd. Transition to None triggers clean up. Handles requests
+    /// affecting the connection status and funding status of a swap.
+    #[display("Swapd Running")]
+    SwapdRunning(SwapdRunning),
+}
+
+pub struct MakeOffer {
+    public_offer: PublicOffer,
+    arb_addr: bitcoin::Address,
+    acc_addr: monero::Address,
+}
+
+pub struct TakerCommit {
+    peerd: ServiceId,
+    public_offer: PublicOffer,
+}
+
+pub struct TakeOffer {
+    public_offer: PublicOffer,
+    arb_addr: bitcoin::Address,
+    acc_addr: monero::Address,
+    peerd: ServiceId,
+}
+
+pub struct SwapdLaunched {
+    peerd: ServiceId,
+    public_offer: PublicOffer,
+    swap_id: SwapId,
+    arbitrating_syncer_up: Option<ServiceId>,
+    accordant_syncer_up: Option<ServiceId>,
+    swapd_up: bool,
+    init_swap: InitSwap,
+    local_trade_role: TradeRole,
+}
+
+pub struct RestoringSwapd {
+    public_offer: PublicOffer,
+    swap_id: SwapId,
+    arbitrating_syncer_up: Option<ServiceId>,
+    accordant_syncer_up: Option<ServiceId>,
+    swapd_up: bool,
+}
+
+pub struct SwapdRunning {
+    peerd: ServiceId,
+    public_offer: PublicOffer,
+    arbitrating_syncer: ServiceId,
+    accordant_syncer: ServiceId,
+    swap_id: SwapId,
+    connected: bool,
+    funding_info: Option<FundingInfo>,
+    auto_funded: bool,
+}
+
+impl StateMachine<Runtime, Error> for TradeStateMachine {
+    fn next(self, event: Event, runtime: &mut Runtime) -> Result<Option<Self>, Error> {
+        match self {
+            TradeStateMachine::StartTaker => transition_to_take_offer(event, runtime),
+            TradeStateMachine::StartMaker => transition_to_make_offer(event, runtime),
+            TradeStateMachine::StartRestore => transition_to_restoring_swapd(event, runtime),
```

Comment:
for consistency:
since they all return `Result<Option<TSM>>` like the other `attempt_transition_to_...`, their return is directly forwarded via `.next()`, and to keep them segregated from the new `transition_to_swapd_launched_tsm`, which actually returns a naked `Result<TSM>`, I'd rename these three  `attempt_transition_to...` as we;;.

Reply:
428c5132925f4f02ed9a3bf6cda34fd7bffd01bf

Reply:
thanks! I was **just** about to comment the same for the syncer state machine ^^